src/dcap_provider.cpp

```diff
@@ -788,13 +807,6 @@ extern "C" void sgx_ql_free_revocation_info(
     delete[] reinterpret_cast<uint8_t*>(p_revocation_info);
 }
 
-extern "C" sgx_plat_error_t sgx_ql_set_base_url(const char* url)
-{
-    cert_base_url = (url == nullptr) ? DEFAULT_CERT_URL : url;

```

Comment:
I'd recommend keeping logic in place default to the default cert url if one isn't specified in the environment variable so we have some control as to where this traffic gets routed to by default.

Reply:
@msft-jonnadul I'm not really sure what you mean if you're talking about something else than what currently exists. I could be misunderstanding of course! 

I see this line at the top of this file:

```
static std::string cert_base_url = DEFAULT_CERT_URL;
```

and then in get_base_url(), there is:

```
const char * env_base_url = getenv("AZURE_BASE_CERT_URL");
if (env_base_url == NULL)
{
    return cert_base_url;
}
```

which will return the default endpoint if the environment variable isn't set. Is that what you mean?

Reply:
Ah my mistake! Didn't see that portion of the refactoring. Yes this should address the issue appropriately. Thanks!